src/wpool_queue_manager.erl

```diff
@@ -30,28 +30,58 @@
          monitors :: #{atom() := monitored_from()},
          queue_type :: queue_type()}).
 
--type state() :: #state{}.
--type from() :: {pid(), reference()}.
+-opaque state() :: #state{}.
+
+-export_type([state/0]).
+
+-type from() :: {pid(), gen_server_reply_tag()}.
+
+-export_type([from/0]).
+
 -type monitored_from() :: {reference(), from()}.
 -type options() :: [{option(), term()}].
+
+-export_type([options/0]).
+
 -type option() :: queue_type.
 -type args() :: [{arg(), term()}].
+
+-export_type([args/0]).
+
 -type arg() :: option() | pool.
 -type queue_mgr() :: atom().
 -type queue_type() :: fifo | lifo.
+-type worker_event() :: new_worker | worker_dead | worker_busy | worker_ready.
+
+-export_type([worker_event/0]).
+
+-type call_request() :: {available_worker, infinity | pos_integer()} | pending_task_count.
 
+-export_type([call_request/0]).
 -export_type([queue_mgr/0, queue_type/0]).
 
+-if(?OTP_RELEASE >= 25).
+
+-type gen_server_request_id() :: gen_server:request_id().
+-type gen_server_reply_tag() :: gen_server:reply_tag().
+
+-else.
+
+-type gen_server_request_id() :: reference().
+-type gen_server_reply_tag() :: reference().
+
+-endif.
+
+-elvis([{elvis_style, export_used_types, disable}]).
+
 %%%===================================================================
 %%% API
 %%%===================================================================
-%% @equiv start_link(WPool, Name, [])
```

Comment:
This whole module is flagged `private` which means `start_link/3` is private, which means we were getting a warning for an `@equiv` for a non-exposed function.